MDAnalysisTests.analysis.test_encore.TestEncore.test_dres_convergence fails randomly

### Expected behaviour
`MDAnalysisTests.analysis.test_encore.TestEncore.test_dres_convergence` should always pass.

### Actual behaviour
`MDAnalysisTests.analysis.test_encore.TestEncore.test_dres_convergence` fails randomly.

Error message:
```
self = <testsuite.MDAnalysisTests.analysis.test_encore.TestEncore object at 0x2aae29ae6e10>, ens1 = <Universe with 3341 atoms>

    def test_dres_convergence(self, ens1):
        expected_values = [ 0.3, 0.]
        results = encore.dres_convergence(ens1, 10)
        assert_almost_equal(results[:,0], expected_values, decimal=1,
>                           err_msg="Unexpected value for Dim. reduction Ensemble similarity in convergence estimation")
E       AssertionError: 
E       Arrays are not almost equal to 1 decimals
E       Unexpected value for Dim. reduction Ensemble similarity in convergence estimation
E       (mismatch 50.0%)
E        x: array([0.5, 0. ])
E        y: array([0.3, 0. ])

testsuite/MDAnalysisTests/analysis/test_encore.py:311: AssertionError
```
### Code to reproduce the behaviour
```bash
#! /bin/bash
cnt=0
exitval=0
while [ $exitval -eq 0 ]; do
    cnt=$(($cnt + 1))
    echo $cnt
    pytest testsuite/MDAnalysisTests/analysis/test_encore.py::TestEncore::test_dres_convergence
    exitval=$?
done
echo "failed after $cnt attempts"
```
When I ran the above bash script, the test passed 31 times before failing.
Test environment:
```
platform linux -- Python 3.5.2, pytest-3.6.1, py-1.5.3, pluggy-0.6.0
plugins: xdist-1.22.2, forked-0.2, hypothesis-3.57.0
```
I also ran the test with Python 2.7.12, where it took 75 runs until it failed. The error message is always the same, btw.

I gave it another shot and it took 452 runs until the test failed, so reproducing the failure might take some time.

### Current version of MDAnalysis
0.18.1-dev

### Problematic code section:
The `dres_convergence()` function implicitly incorporates stochastic elements employing the `rand()` function from the C header `stdlib.h`. The PRNG is seeded [here](https://github.com/MDAnalysis/mdanalysis/blob/9675f920f440e88c540fbf91c98b162737604a1f/package/MDAnalysis/analysis/encore/dimensionality_reduction/src/spe.c#L196):
>     srand(time(NULL)+getpid()*getpid());

I guess having the opportunity to supply a fixed random seed for testing purposes wouldn't help much, since what's behind `rand()` is implementation-defined.
Maybe simply allowing the test to fail *once* and if so, repeating it a couple of times mitigates the problem.

--- a/package/MDAnalysis/analysis/encore/dimensionality_reduction/src/spe.c
+++ b/package/MDAnalysis/analysis/encore/dimensionality_reduction/src/spe.c
@@ -193,10 +193,8 @@ double CStochasticProximityEmbedding(
     double t = 0.0;
     double finalstress = 0.0;
 
-    srand(time(NULL)+getpid()*getpid());
     /* random init of d */
-
-
+    srand(time(NULL)+getpid()*getpid());
     for (int i=0; i<nelem*dim; i++) {
         d_coords[i] = (double) rand() / (double) RAND_MAX;
     }
@@ -228,8 +226,8 @@ double CStochasticProximityEmbedding(
             }
         }
         lam = lam - (maxlam - minlam) / (double)(ncycle - 1);
-        if (i % stressfreq == 0 && i != 0 && stressfreq > 0)
-	  printf("Cycle %d - Residual stress: %.3f, lambda %.3f\n", i, neighbours_stress(s, d_coords, dim, nelem, rco),lam);
+        //if (i % stressfreq == 0 && i != 0 && stressfreq > 0)
+	    //printf("Cycle %d - Residual stress: %.3f, lambda %.3f\n", i, neighbours_stress(s, d_coords, dim, nelem, rco),lam);
     }
     finalstress = neighbours_stress(s, d_coords, dim, nelem, rco);
     //printf("Calculation finished. - Residual stress: %.3f\n", finalstress);

--- a/testsuite/MDAnalysisTests/analysis/test_encore.py
+++ b/testsuite/MDAnalysisTests/analysis/test_encore.py
@@ -305,10 +305,27 @@ inconsistent results")
                                 err_msg="Unexpected value for Clustering Ensemble similarity in convergence estimation")
 
     def test_dres_convergence(self, ens1):
-        expected_values = [ 0.3, 0.]
+        # Due to encore.dres_convergence() involving random numbers, the
+        # following assertion is allowed to fail once. This significantly
+        # reduces the probability of a random test failure.
+        expected_values = [0.3, 0.]
         results = encore.dres_convergence(ens1, 10)
-        assert_almost_equal(results[:,0], expected_values, decimal=1,
-                            err_msg="Unexpected value for Dim. reduction Ensemble similarity in convergence estimation")
+        try:
+            assert_almost_equal(results[:,0], expected_values, decimal=1)
+        except AssertionError:
+            # Random test failure is very rare, but repeating the failed test
+            # just once would only assert that the test passes with 50%
+            # probability. To be a little safer, we raise a warning and repeat
+            # the test 10 times:
+            warnings.warn(message="Test 'test_dres_convergence' failed, "
+                                  "repeating test 10 times.",
+                          category=RuntimeWarning)
+            for i in range(10):
+                results = encore.dres_convergence(ens1, 10)
+                assert_almost_equal(results[:,0], expected_values, decimal=1,
+                                    err_msg="Unexpected value for Dim. "
+                                            "reduction Ensemble similarity in "
+                                            "convergence estimation")
 
     def test_hes_error_estimation(self, ens1):
         expected_average = 10

